blackbird_python/blackbird/auxiliary.py

```diff
@@ -201,7 +201,16 @@ def _expression(expr):
                 )
             )
 
-        return _VAR[expr.getText()]
+        name = expr.getText()
+        # check if expr is a p-type parameter; it must be declared as an array.
```

Comment:
What is a p-type parameter? :thinking: would this be the allowed parameter names for tdm? (with the updated names)